### PR TITLE
Change UF ho to ppRewrite instead of expand definition

### PR DESCRIPTION
**Pull request by ajreynol**
UF uses expandDefinitions to convert fully applied HO_APPLY to APPLY_UF.  The more appropriate place to do this is in Theory::ppRewrite.

### CODE REVIEW

File: test/regress/regress1/ho/NUM925^1.p [hunk 1]
```diff
@@ -1,4 +1,4 @@
-% COMMAND-LINE:  --uf-ho
+% COMMAND-LINE:  --uf-ho --ho-elim
```
Comment (HanielB) [inline test/regress/regress1/ho/NUM925^1.p:1]: Why did you need to add the option?

Comment (ajreynol) [reply test/regress/regress1/ho/NUM925^1.p:1]: Because we answer unknown otherwise.

Comment (HanielB) [reply test/regress/regress1/ho/NUM925^1.p:1]: But we were solving them without the pass before? Why is that?

Comment (ajreynol) [reply test/regress/regress1/ho/NUM925^1.p:1]: This appears to be random.

Comment (HanielB) [reply test/regress/regress1/ho/NUM925^1.p:1]: I see.

**Review by HanielB — APPROVED** [review]
